### PR TITLE
fix: icon_value roundup in solana

### PR DESCRIPTION
**Pull request by Itshyphen**
### Commit Message

```bash
fix: icon_value roundup in solana
```

see the [guidelines](https://github.com/icon-project/community/blob/main/guidelines/technical/software-development-guidelines.md#commit-messages) for commit messages.

### Changelog Entry

```bash
version: <log entry>
```

## Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have documented my code in accordance with the [documentation guidelines](https://github.com/icon-project/community/blob/main/guidelines/technical/software-development-guidelines.md#documentation)
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have run the unit tests
- [ ] I only have one commit (if not, squash them into one commit).
- [ ] I have a descriptive commit message that adheres to the [commit message guidelines](https://github.com/icon-project/community/blob/main/guidelines/technical/software-development-guidelines.md#commit-messages)
- [ ] I have added version bump label on PR.

> Please review the [CONTRIBUTING.md](/CONTRIBUTING.md) file for detailed contributing guidelines.

### CODE REVIEW

**Review by ibrizsabin — APPROVED** [review]

File: programs/balanced-dollar/src/instructions.rs [hunk 1]
```diff
@@ -51,11 +51,26 @@ pub fn cross_transfer<'info>(
     ctx: Context<'_, '_, '_, 'info, CrossTransfer<'info>>,
     to: String,
     value: u64,
+    icon_bnusd_value: Option<u64>,
```
Comment (AntonAndell) [inline programs/balanced-dollar/src/instructions.rs]: Shouldn't this be u128?

File: programs/balanced-dollar/src/instructions.rs [hunk 2]
```diff
@@ -51,11 +51,26 @@ pub fn cross_transfer<'info>(
     ctx: Context<'_, '_, '_, 'info, CrossTransfer<'info>>,
     to: String,
     value: u64,
```
Comment (AntonAndell) [inline programs/balanced-dollar/src/instructions.rs]: Do we even need this when we have ctx.accounts.from.amount?

**Review by AntonAndell — CHANGES_REQUESTED** [review]

File: programs/balanced-dollar/src/instructions.rs (hunk 2 above)
Comment (gcranju) [reply programs/balanced-dollar/src/instructions.rs]: this is the total amount user hold,  was thinking, there should be limit how much user want to spend?

Comment (AntonAndell) [reply programs/balanced-dollar/src/instructions.rs]: Yeah okey, we could potentially still opt to only have ICON_bnUSD_value

Comment (gcranju) [reply programs/balanced-dollar/src/instructions.rs]: yeah done

**Review by AntonAndell — APPROVED** [review]
I like this for tokens that have to change decimals 👍

**Pull request merged**